src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/UseExceptionThrowHelpersTests.cs

```diff
@@ -0,0 +1,1078 @@
+        ArgumentOutOfRangeException.ThrowIfNegativeOrZero(arg);
+
+        ArgumentOutOfRangeException.ThrowIfLessThanOrEqual(arg, 42);
+        ArgumentOutOfRangeException.ThrowIfLessThanOrEqual(arg, 42);
+
+        ArgumentOutOfRangeException.ThrowIfLessThan(arg, 42);
+        ArgumentOutOfRangeException.ThrowIfLessThan(arg, 42);
+
+        ArgumentOutOfRangeException.ThrowIfGreaterThan(arg, 42);
+        ArgumentOutOfRangeException.ThrowIfGreaterThan(arg, 42);
+
+        ArgumentOutOfRangeException.ThrowIfGreaterThanOrEqual(arg, 42);
+        ArgumentOutOfRangeException.ThrowIfGreaterThanOrEqual(arg, 42);
+
+        ArgumentOutOfRangeException.ThrowIfGreaterThan(arg, (int)TimeSpan.FromSeconds(42).TotalSeconds);
+        ArgumentOutOfRangeException.ThrowIfGreaterThan(arg, (int)TimeSpan.FromSeconds(42).TotalSeconds);
+
+        if (arg is 42)
+            throw new ArgumentOutOfRangeException(nameof(arg));
+
+        if (arg == 42)
+            throw new ArgumentOutOfRangeException(nameof(arg));
+
+        if (arg is 0)
+        {
+            Console.WriteLine();
+            throw new ArgumentOutOfRangeException(nameof(arg));
+        }
+
+        if (arg is < 0) // we could augment the analyzer in the future to support this
+        {
+            throw new ArgumentOutOfRangeException(nameof(arg));
+        }
+
+        if (arg > 42 && arg < 84) // we could augment the analyzer in the future to support this
+            throw new ArgumentOutOfRangeException(nameof(arg));
+    }
+}
+"
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task ObjectDisposedExceptionThrowIf_DoesntExist_NoDiagnostics()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+
+namespace System
+{
+    public class ObjectDisposedException : Exception
+    {
+        public ObjectDisposedException(string type) { }
+        public ObjectDisposedException(string type, string message) { }
+    }
+}
+
+class C
+{
+    private bool IsDisposed { get; set; }
+
+    void M()
+    {
+        if (IsDisposed) throw new ObjectDisposedException(null);
+
+        if (IsDisposed)
+            throw new ObjectDisposedException(GetType().Name);
+
+        if (IsDisposed)
+        {
+            throw new ObjectDisposedException(GetType().FullName);
+        }
+
+        if (DateTime.UtcNow.Hour == 0)
+        {
+            throw new ObjectDisposedException(nameof(DateTime));
+        }
+    }
+
+    string Prop
+    {
+        get
+        {
+            if (IsDisposed) throw new ObjectDisposedException(null);
+            return ""test"";
+        }
+    }
+}
+");
+        }
+
+        [Fact]
+        public async Task ObjectDisposedExceptionThrowIf_FixesAppliedAsAppropriate()
+        {
+            var test = new VerifyCS.Test()
+            {
+                TestCode = @"
+using System;
+
+namespace System
+{
+    public class ObjectDisposedException : Exception
+    {
+        public ObjectDisposedException(string type) { }
```

Comment:
should it test that the other ODE 2 ctors are ignored?

Reply:
We could add additional tests.  The tests are written knowing that the same paths are used to handle all the exception types (other than ArgumentException with its weird parameter ordering), so the ArgumentNullException tests have tests for other arities.